mono/utils/w32handle.c

```diff
@@ -1346,20 +1346,31 @@ mono_w32handle_wait_multiple (gpointer *handles, gsize nhandles, gboolean waital
 
 				if (i < lowest)
 					lowest = i;
+
+				if (!waitall)
```

Comment:
I would side more on keeping the same behaviour as before, except if we have an explicit test that shows that on .NET they only return on the first one (which sounds weird to be honest)